Warning for photomet added

ladoramkershner opened the pull request:
Added warning if 'DEM' angle source option is used with 'mixed' or 'topo' normalization method

## Description
A warning is issued to the application logs but still outputs result cube. This warning style was chosen because some information as to why the angle source/normalization methods cannot be paired is not clearly outlined. We do not want to prevent users who understand the fundamentals behind the photomet methods from making an educated decision to couple 'DEM' and 'Mixed'/'Topo'.

## Related Issue
Closes #3451 and #3452. 

## Motivation and Context
There is some lost information within the building as to why this change is necessary, however, it is based on tickets #3451 and #3452, which were migrated from redmine. 

## How Has This Been Tested?
New tests for photomet were added to test that the new warning is appropriately output. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- - [ ] My code follows the code style of this project. -->
- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

Commit:
added warning if anglesource option is used with mixed or topo normalization method; based on github tickets 3451 and 3452

Commit:
adding tests

krlberry reviewed (dismissed):
:+1: Looks good!

 I'm honestly not sure if we're still doing this or not, but in the past, for changes to ISIS applications, we would add a history entry which explains the change that was made to the documentation xml file associated with the ISIS application. (In this case, `photomet.xml`) I have an example of this in my PR for `voy2isis` #3490.

I'm approving the PR anyway since I'm not sure whether this is necessary or not.

jessemapel commented:
@kberryUSGS That is still our standards

Kelvinrr commented:
Looks good to me, will approve after history comment is added.

krlberry commented:
@jessemapel :+1: Thanks. It had been so long since I'd actually updated an ISIS application that I couldn't remember.

scsides commented on isis/src/base/apps/photomet/main.cpp:
```diff
@@ -1758,6 +1758,12 @@ void IsisMain() {
+    PvlGroup warning("Warning");
+    warning.addKeyword(PvlKeyword("Warning",message),Pvl::Replace);
+    Application::Log(warning);
+  }
```
Using Pvl::Replace will overwrite any other "Warning" keywords in a group. Yes the group was just created, so why add the Replace at all? If additional tests similar to this one are added they should add an additional "Warning" keyword to this group. With the "Replace" only one will show in the log.

scsides replied:
Suggest changing the group name to "Warnings"

ladoramkershner replied:
I changed the pvlgroup name to Warnings and removed the Pvl::Replace.

scsides commented on isis/src/base/apps/photomet/tsts/testDemWarning/Makefile:
```diff
@@ -0,0 +1,31 @@
+APPNAME = photomet
+
+angleSourceDemUsedemFalse.cub.TOLERANCE = 0.001
+angleSourceDemUsedemTrue.cub.TOLERANCE = 0.001
```
These look like leftovers from a cut-n-paste. Should be removed

Commit:
Addressing Stuarts comments

Commit:
Addressing Kristin's comment

ladoramkershner commented:
>  Looks good!
> 
> I'm honestly not sure if we're still doing this or not, but in the past, for changes to ISIS applications, we would add a history entry which explains the change that was made to the documentation xml file associated with the ISIS application. (In this case, `photomet.xml`) I have an example of this in my PR for `voy2isis` #3490.
> 
> I'm approving the PR anyway since I'm not sure whether this is necessary or not.

@kberryUSGS @jessemapel The entirety of photomet.xml is getting reported in the file change section of this PR. So to hopefully save some sanity, in response to your comments I added:
```
<change name="Lauren Adoram-Kershner" date="2019-11-07">
    Added a warning when the 'DEM' angle source option is used with 'mixed'
    or 'topo' normalization method. Fixes #3451 and #3452.
</change>
```
at the end of the <history> section. I know this is vague, but because there was not a lot of explanatory information surrounding the request for this change, I was not sure what to put. If you have any suggestions, I am willing to add more.

jessemapel commented:
> @kberryUSGS @jessemapel The entirety of photomet.xml is getting reported in the file change section of this PR.

If you click on the little gear you can hide whitespace changes like that. It has to do with slick edit adding spaces and atom removing them.

scsides reviewed (approved):
Looks good
